### DISCUSSION TITLE
تانل پنل به پنل با مرزبان

### DISCUSSION THREAD

**Discussion by moonjom** [Gozargah/Marzban#473] [General]
**در این آموزش نحوه تانل بین دو سرور ایران و خارج را در مرزبان قصد داریم آموزش دهیم . شما به یک سرور خارج و یک سرور ایران برای این کار نیاز دارید .**

ابتدا بر روی سرور خارج پنل مرزبان را نصب می‌کنیم و سپس سرور ایران را به صورت **نود** به سرور خارج متصل می‌کنیم.
برای نصب مرزبان نود در صورتی که خطا گرفتید می‌توانید از دی ان اس [شکن](https://shecan.ir/) استفاده کنید.

بعد از اتمام راه اندازی پنل و نود شما نیاز دارید یک اینباند برای سرور ایران‌ و یک اینباند دیگر برای سرور خارج تعریف کنید.ما در این آموزش از تانل ریلیتی به ریلیتی استفاده می‌کنیم ولی شما میتوانید از هر پروتکلی برای این کار استفاده کنید .

بعد از انجام این مراحل یک یوزر نامحدود با سرور خارج میسازیم و پس از تست درستی کارکرد کدهای json این یوزر را به وسیله کلاینت هایی مثل v2rayng و ... بدست می‌آوریم و در مکانی ذخیره می‌کنیم.

حالا کاری که ما باید انجام بدهیم این است که اینباند ایران رو به اوت باند سرور خارج بفرستیم . برای این‌کار در قسمت اینباند که پروتکل مورد نظر را قرار می‌دهیم ‌, در قسمت اوت باند کدهای json یوزر نامحدودی که در مرحله قبل را ساختیم قرار می‌ دهیم و در مرحله آخر در قسمت روتینگ مسیر اینباند سرور ایران رو از طریق اوت باند سرور خارج عبور می‌ دهیم.

**برای یادگیری راحت تر فایل های زیر رو قرار دادیم :**

[inbounds.txt](https://github.com/Gozargah/Marzban/files/12369652/inbounds.txt)

[outbands.txt](https://github.com/Gozargah/Marzban/files/12369655/outbands.txt)

**دقت کنید اوت‌باند سرور خارج باید در دومین اوت‌باند به صورت فایل بالا باید قرار بگیرد .**

[routing.txt](https://github.com/Gozargah/Marzban/files/12369657/routing.txt)

** آخرین مرحله کار ما هم به صورت فایل بالا می‌باشد که به وسیله روتینگ اینباند سرور ایران را به اوت باند سرور خارج متصل می‌کنیم.**
![22](https://github.com/Gozargah/Marzban/assets/122884265/51a33390-cd36-4fa3-bebb-26c177328344)

**Comment by hva8** [comment]
این فایلها رو باید کجا ذخیره کنیم؟

**Reply by mozhutte** [reply]
سلام.
طبیعتا شما اگر از مرزبان استفاده می‌کنید با نحوه‌ی اضافه‌کردن اینباند و کلا کار کردن با فایل config آشنا هستید. این فایل‌ها رو طبق اسم فایل‌ها باید متناسب با نیاز خودتون به بخش‌های اینباند و اوتباند و روتینگ core settings اضافه کنید

**Reply by MrLlnux** [reply]
سلام کد برای vmess tcp هم لطف میکنید قرار بدهید با تشکر

**Reply by Ramintn** [reply]
سلام من سایت های ایران رو بستم و وارپ هم دارم که اسپاتیفای رو از وارپ رد کردم روتینگ رو چجوری باید تغییر بدم؟ ضمنا تو قسمت اوتباندز که نوشته ایپی شما ،باید ایپی سرور خارج رو قرار بدیم یا ایران رو؟

**Comment by uRepow** [comment]
من همه رو به درستی انجام دادم ولی تانل نشد و روی همون ایپی ابران مونده برای سرور ایران هم باید پنل باز کنیم یا خیر ممنون میشم واضح تر توضیح بدید

**Comment by KAJOOSH** [comment]
اگر خواسته باشیم ترافیک از سرور  اصلی خارج که پنل هست رد نشه و از یکی از نود های خارج عبور کند میشود؟

**Reply by AlamiShayan** [reply]
راه حلی برای تانل کردن نودها پیدا کردید؟من خیلی گشتم اگر چیزی هست ممنون میشم بفرمایید

**Comment by nasiriscript** [comment]
دمت گرم عالیه فکر نمیکردم بشه سرور ایران رو نود کرد

**Comment by MrOtto94** [comment]
امکانش هست آموزش ویدیویی برای این مدل تانل بسازید ؟
فک میکنم نیاز خیلی ها باشه

**Reply by w2er416tgjd** [reply]
vless://0a05edde-aa2b-417e-90ff-4e7ef976b52b@hidden:3810?security=reality&type=tcp&headerType=&path=&host=&sni=google.com&fp=chrome&pbk=SbVKOEMjK0sIlbwg4akyBg5mL5KZwwB-ed4eEE7YnRc&sid=6ba85179e30d4fc2#%F0%9F%94%8C%20%F0%9F%93%B6%200%20B/30.0%20GB%20%E2%8C%9B29%20%20Days vless://0a05edde-aa2b-417e-90ff-4e7ef976b52b@developers.cloudflare.com:2052?security=none&type=httpupgrade&headerType=&path=%2F%3Fed%3D2560&host=notlden.nextnumber.ir#%F0%9F%87%A9%F0%9F%87%B0%20OTHERS ss://Y2hhY2hhMjAtaWV0Zi1wb2x5MTMwNTp2Q3lPZ2tpNDhhRWd5eEZSR1p2Tzh3@[2a05:541:123:2f::1]:2060#%F0%9F%87%A9%F0%9F%87%B0%20Rightel%20%26%20Mtn%20%26%20Mci%20%5Bipv6%5D vless://0a05edde-aa2b-417e-90ff-4e7ef976b52b@89.42.138.222:2074?security=none&type=tcp&headerType=http&path=%2F&host=4d36199915d0c1d5.iran.ir#%F0%9F%87%A9%F0%9F%87%B0%20WIFI

**Comment by mahsamcse** [comment]
دوست عزیز
منم اصلا متوجه نشدم و نتونستم پیدا کنم حتی با فایل های خوده شما
یکم توضیحات گنگ و خلاصه هست بعضی جاهاش
ممنون میشم یا کامل تر کنید همینجا و یا یک ویدئو درست کنید
توی این شرایط شدت گرفتم فیلترینگ و عدم کارایی روش های دیگه تانل حتما این روش بسیار کمک کننده خواهد بود
با تشکر از زحمت شما

**Comment by milad73n** [comment]
مشکل این روش اینه که وقتی اوتباند رو میزاریم اصلا به روتینگ توجه نمیکنه و تمام اینباندامونو از اوتباند رد میکنه
راه حلی هست؟

**Comment by m0x61h0x64i** [comment]
اکسس نمیشه؟

**Comment by AliRMT** [comment]
سلام ممنون میشم اگر برای این مورد یک ویدئو آموزشی قرار بدید. اونجوری ک متوجه شدم بعضی ها هم مثل ما مشکل دارن و متوجه نشدن.

اگر یک ویدئو بزنید عالی میشه

**Comment by Eslender73** [comment]
پنل به پنل ریورس نمیشه زد؟
